src/components/AppProvider/types.ts

```diff
@@ -1,8 +1,34 @@
 import {ClientApplication} from '@shopify/app-bridge';
 import {ValidationMap} from 'react';
 import * as PropTypes from 'prop-types';
-import {THEME_CONTEXT_TYPES as polarisTheme} from '../ThemeProvider';
-import {Intl, Link, StickyManager, ScrollLockManager} from './utilities';
+import {LinkLikeComponent} from '../UnstyledLink';
+import {Theme, THEME_CONTEXT_TYPES as polarisTheme} from '../ThemeProvider';
+import {
+  Intl,
+  Link,
+  StickyManager,
+  ScrollLockManager,
+  TranslationDictionary,
+} from './utilities';
+
+export interface AppProviderProps {
+  /** A locale object or array of locale objects that overrides default translations */
+  i18n?: TranslationDictionary | TranslationDictionary[];
+  /** A custom component to use for all links used by Polaris components */
+  linkComponent?: LinkLikeComponent;
+  /** The API key for your application from the Partner dashboard */
+  apiKey?: string;
+  /**
+   * The current shop’s origin, provided in the session from the Shopify API (to be provided without the https://)
+   * @default getShopOrigin()
```

Comment:
Unless I'm mistaken I don't think our parser can handle this?

Reply:
I believe you're right, the default stuff can only handle strings ATM, but this is how it was before. I've only moved the definitions around, I haven't modified them so I'm going to claim fixing this up is out of scope for this PR :)